libs/coin-evm/src/api/gasTracker/ledger.ts

```diff
@@ -54,30 +54,40 @@ export const getGasOptions = async ({
       useEIP1559 ? "?display=eip1559" : ""
     }`,
   }).then(({ data }) => ({
-    low: new BigNumber(data.low),
-    medium: new BigNumber(data.medium),
-    high: new BigNumber(data.high),
-    next_base: new BigNumber(data.next_base),
+    /**
+     * Note: the explorer API is supposed to return integer values for each field,
+     * but since we can't guarantee that, better be safe than sorry
+     */
+    low: new BigNumber(data.low).integerValue(),
+    medium: new BigNumber(data.medium).integerValue(),
+    high: new BigNumber(data.high).integerValue(),
+    next_base: new BigNumber(data.next_base).integerValue(),
   }));
 
   const EIP1559_BASE_FEE_MULTIPLIER: number = getEnv("EIP1559_BASE_FEE_MULTIPLIER");
 
+  /**
+   * Since our use of BigNumber implies only using integers, we need to round up to
```

Comment:
we do when we parse an amount from the user / format an amount to the user (so on the two extremes of the spectrum)
but most of the time, unless there are intermediary calculations (like done here, we can multiply/divide, in the realm of floating numbers) we will eventually need to go back to integers since we usually store everything on the atomic level (we can't divide a satoshi / a wei)